### ISSUE TITLE
Variational Autoencoders Note

### ISSUE THREAD

**Issue by rockey1006** [rockey1006/Variational-Autoencoders#1]
Variational Autoencoders Learning Note
Ruoqi Wei
	Intuitively Understanding 
![image](https://user-images.githubusercontent.com/15843177/46185760-5f5d1f00-c2a9-11e8-92dd-4e5e05f5472c.png)
![image](https://user-images.githubusercontent.com/15843177/46185775-697f1d80-c2a9-11e8-845e-9fbcf0d48371.png)

1.	Model: Probability distribution Generative model that generate a random, new output, that looks similar to the training data.（reconstruct original data）
2.	Effect of Encoders: Dimensionality reduction, Incres, missing partial filling for better identification
3.	An encoder network takes in an input, and converts it into a smaller, dense representation, which the decoder network can use to convert it back to the original input.
In computer vision,it can be used to initialize a supervised model with better features as a Fine-tune encoder jointly with classifier. An autoencoder network is actually a pair of two connected networks, an encoder and a decoder. 

 
(Note: The generation method is based on the data learning joint probability distribution P(X, Y), and then the conditional probability distribution P(Y|X) is obtained as a prediction model, which is called a generative model:
![image](https://user-images.githubusercontent.com/15843177/46185808-8a477300-c2a9-11e8-8557-c13a06500fdd.png)

 )
 

  
![image](https://user-images.githubusercontent.com/15843177/46185819-93384480-c2a9-11e8-9032-d1181661606d.png)

4.	Why VAEs?
When training a generative model, the more complicated the dependencies between the dimensions, the more difficult the models are to train. The autoencoder takes in an input and produces a much smaller representation that contains enough information for the next part of the network, so that making the model easier to train.
Compare to standard autoencoder,what’s the advantages of VAEs?

The problem with standard autoencoders is that the latent space may not be continuous, or allow easy interpolation. If the space has discontinuities and you generate a variation from there, the decoder will simply generate an unrealistic output. The VAEs Improved this shortcoming by making the latent spaces  continuous and allowing easy random sampling and interpolation.

5.	How VAEs achieved?
![image](https://user-images.githubusercontent.com/15843177/46185836-a519e780-c2a9-11e8-898c-bc1ca577bd7c.png)

	It achieves this by outputting two vectors of size n: a vector of means, μ, and another vector of standard deviations, σ. 
 
![image](https://user-images.githubusercontent.com/15843177/46185844-ae0ab900-c2a9-11e8-8ac9-d82557cf2260.png)

	They form the parameters of a vector of random variables of length n, with the i th element of μ and σ being the mean and standard deviation of the ith random variable, Xi, from which we sample, to obtain the sampled encoding which we pass onward to the decoder: 
 
![image](https://user-images.githubusercontent.com/15843177/46185849-b7942100-c2a9-11e8-95ee-90338d8dd03d.png)

	This stochastic generation means, that even for the same input, while the mean and standard deviations remain the same, the actual encoding will somewhat vary on every single pass simply due to sampling.
![image](https://user-images.githubusercontent.com/15843177/46185857-c084f280-c2a9-11e8-9d03-7f5c17853780.png)

 
![image](https://user-images.githubusercontent.com/15843177/46185864-c975c400-c2a9-11e8-8c94-18f263a01bec.png)

![image](https://user-images.githubusercontent.com/15843177/46185877-d2ff2c00-c2a9-11e8-9381-08e1d1a85d30.png)

	The mean vector controls where the encoding of an input should be centered around, while the standard deviation controls the “area”.
 
	What we ideally want are encodings, all of which are as close as possible to each other while still being distinct, allowing smooth interpolation, and enabling the construction of new samples.In order to force this, we introduce the Kullback–Leibler divergence (KL divergence) into the loss function. 
	Minimizing the KL divergence here means optimizing the probability distribution parameters (μ and σ) to closely resemble that of the target distribution.
   

References:
	https://towardsdatascience.com/intuitively-understanding-variational-autoencoders-1bfe67eb5daf
	https://wiseodd.github.io/techblog/2016/12/10/variational-autoencoder/
	https://medium.com/@cotra.marko/making-sense-of-the-kullback-leibler-kl-divergence-b0d57ee10e0a
	Kingma D P, Welling M. Auto-Encoding Variational Bayes[J]. stat, 2014, 1050: 10.
	DOERSCH C. Tutorial on Variational Autoencoders[J]. stat, 2016, 1050: 13.
	Mahmood, Ausif,” Variational Autoencoders.”Lecture of Deep Learning course from University of Bridgeport.
	Feifei,Li, “Generative Models.”Lecture of CS231n: Convolutional Neural Networks for Visual Recognition from Stanford University.
	Blei, David M., "Variational Inference." Lecture of Advanced Methods in Probabilistic Modeling course from Princeton University.
